test/middleware/request.js

```diff
@@ -1,98 +1,118 @@
 import test from 'ava';
 import nock from 'nock';
+import sinon from 'sinon';
+import rewire from 'rewire';
 
 import { CALL_API } from '../../app/src/js/actions/types';
-import { requestMiddleware } from '../../app/src/js/middleware/request';
+const request = rewire('../../app/src/js/middleware/request');
 
 const port = process.env.FAKEAPIPORT || 5001;
+const token = 'fake-token';
 
-test.beforeEach((t) => {
-  const doDispatch = () => {};
-  const doGetState = () => ({
-    api: {
-      tokens: {
-        token: 'fake-token'
-      }
+const requestMiddleware = request.__get__('requestMiddleware');
+const loginErrorStub = sinon.stub();
+request.__set__('loginError', () => {
+  loginErrorStub();
+});
+
+const getStateStub = sinon.stub().callsFake(() => ({
+  api: {
+    tokens: {
+      token
     }
-  });
-  t.context.nextHandler = requestMiddleware({dispatch: doDispatch, getState: doGetState});
+  }
+}));
+const dispatchStub = sinon.stub();
+const nextStub = sinon.stub();
+
+const create = () => {
```

Comment:
Since I was updating the request middleware, I needed to add tests. While I was in here, it made sense to update the testing pattern to match the Redux docs:

https://redux.js.org/recipes/writing-tests#middleware

I also think the tests are a lot more readable than they were before, but I guess it's subjective

Reply:
This is nice! 👍